File: benchmarks/benchmark.pl
use strict;
use warnings;
use Time::HiRes qw(time);
sub fibonacci { my ($number) = @_; return $number if $number < 2; return fibonacci($number - 1) + fibonacci($number - 2); }
sub count_primes {
    my ($limit) = @_; my @is_prime = (1) x ($limit + 1); $is_prime[0] = $is_prime[1] = 0;
    for my $number (2 .. int(sqrt($limit))) { next unless $is_prime[$number]; for (my $multiple = $number * $number; $multiple <= $limit; $multiple += $number) { $is_prime[$multiple] = 0; } }
    my $count = 0; $count += $_ for @is_prime; return $count;
}
my $fibonacci_input = $ARGV[0] // 35; my $prime_limit = $ARGV[1] // 200000;
my $start = time(); my $fibonacci_result = fibonacci($fibonacci_input); my $fibonacci_time = time() - $start;
$start = time(); my $prime_result = count_primes($prime_limit); my $prime_time = time() - $start;
printf "%.6f,%.6f\n", $fibonacci_time, $prime_time;

File: benchmarks/benchmark.rs
use std::time::Instant;

fn fibonacci(number: i32) -> i64 {
    if number < 2 { return i64::from(number); }
    fibonacci(number - 1) + fibonacci(number - 2)
}

fn count_primes(limit: usize) -> usize {
    let mut is_prime = vec![true; limit + 1];
    is_prime[0] = false;
    is_prime[1] = false;
    for number in 2..=((limit as f64).sqrt() as usize) {
        if is_prime[number] {
            for multiple in (number * number..=limit).step_by(number) { is_prime[multiple] = false; }
        }
    }
    is_prime.into_iter().filter(|value| *value).count()
}

fn main() {
    let arguments: Vec<String> = std::env::args().collect();
    let fibonacci_input = arguments.get(1).and_then(|value| value.parse().ok()).unwrap_or(35);
    let prime_limit = arguments.get(2).and_then(|value| value.parse().ok()).unwrap_or(200000);
    let start = Instant::now(); let fibonacci_result = fibonacci(fibonacci_input); let fibonacci_time = start.elapsed().as_secs_f64();
    let start = Instant::now(); let prime_result = count_primes(prime_limit); let prime_time = start.elapsed().as_secs_f64();
    println!("{:.6},{:.6}", fibonacci_time, prime_time);
}
